src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -488,14 +496,86 @@ public FormValidation doCheckTargetCredentialsApiKey(
         }
         if (CredentialsProvider.listCredentials(StringCredentials.class,
                         item,
-                        ACL.SYSTEM, 
+                        ACL.SYSTEM,
                         Collections.emptyList(),
                         CredentialsMatchers.withId(targetCredentialsApiKey)).isEmpty()) {
             return FormValidation.error("Cannot find currently selected credentials");
         }
         return FormValidation.ok();
     }
 
+    /**
+     * Populates the targetCredentialsApplicationKey field from the configuration screen with all of the valid credentials
+     *
+     * @param item - The context within which to list available credentials
+     * @param targetCredentialsApplicationKey - A String containing the application key as a credential
```

Comment:
Are there length checks before the string params gets here? I.e. can someone throw in a 2,147,483,647 char key?

Reply:
The `String` object is created by the Java Servlet mechanism that parses request parameters. This is outside of the plugin scope, we just get whatever is provided by the lower abstraction layers.

This is being submitted in a POST request, so there is no limit on the length of parameter. It is technically possible for someone to submit a parameter this long. It would create a 4GB string in the JVM heap, which either occupies a significant part of it, or lead to an `OutOfMemoryError` (in a random place in the Jenkins process, does not have to be the plugin) if the heap is not big enough

Reply:
In this regard it's identical to any other user-supplied config parameter

Reply:
Thanks.  It sounds like there are no unique vulnerabilities the this would be subject to in terms of inputs, and likely a failure would already occur out of scope much further upstream.  Is it safe to say then that even though the APM tracer configuration is ingesting new inputs, it is not responsible for handling the sanitization and enforcing limits, delegating that to the servlet?

Reply:
Correct, request parameter parsing is done by servlet.